core/shared/src/main/scala/com/lynbrookrobotics/potassium/PeriodicSignal.scala

```diff
@@ -125,6 +125,29 @@ abstract class PeriodicSignal[+T] { self =>
     }
   }
 
+  /**
+    * Applies a timestamped fixed size sliding window over the signal
+    * @param size the size of the window
+    * @param filler the element to use to fill the window until elements are available
+    * @return
+    */
+  def slidingWithTime[U >: T](size: Int, filler: U): PeriodicSignal[Queue[(Time, U)]] = {
```

Comment:
You should probably simplify the job of this into being a `zipWithTime`, so that you can do `signal.zipWithTime.sliding(...)`.